extension/azureauthextension/extension.go

```diff
@@ -25,6 +25,7 @@ import (
 type authenticator struct {
 	credential azcore.TokenCredential
 	logger     *zap.Logger
+	config     *Config
```

Comment:
Instead of adding `config`, can you instead just add `scopes`? So we don't have to carry everything from configuration around, as we don't need the other variables

Reply:
done!